File: app/src/main/java/com/example/ugcssample/drone/DroneBridge.java
package com.example.ugcssample.drone;

import com.example.ugcssample.utils.PermissionCheckResult;

import dji.sdk.products.Aircraft;

public interface DroneBridge {

    void submitLocationInit();

    void submitSdkInit();

    void onUsbAccessoryAttached();

    PermissionCheckResult getPermissionCheckResult();

    Aircraft getAircraftInstance();

    void startModelSimulator();

    void setMediaFileCustomInformation(String information);

    void onDestroy();

    void remoteController();

    void zoomCamera();

    void wideCamera();

    void remoteControllerBind();
}

File: app/src/main/java/com/example/ugcssample/drone/DroneBridge.kt
package com.example.ugcssample.drone

import android.content.Context
import com.example.ugcssample.utils.PermissionCheckResult
import dji.sdk.products.Aircraft
import java.io.File

interface DroneBridge {
    fun submitLocationInit()
    fun submitSdkInit()
    fun onUsbAccessoryAttached()
    val permissionCheckResult: PermissionCheckResult?
    val aircraftInstance: Aircraft?
    fun startModelSimulator()
    fun onDestroy()
    fun takeOff()
    fun land(useKeyInterface: Boolean)
    fun uploadDemoMission()
    fun startMission()
    fun uploadMission(nativeRoute : File)
    fun cancelLanding()
}